kDrive/UI/Controller/Files/FileActionsFloatingPanelViewController+Actions.swift

```diff
@@ -197,9 +197,16 @@ extension FileActionsFloatingPanelViewController {
         } else {
             // Create share link
             setLoading(true, action: action, at: indexPath)
-            Task { [proxyFile = file.proxify()] in
+            Task { [frozenFile = file.freezeIfNeeded()] in
+                let proxyFile = frozenFile.proxify()
                 do {
-                    let shareLink = try await driveFileManager.createShareLink(for: proxyFile)
+                    // TODO: Optimise API calls
+                    if frozenFile.hasSharelink {
```

Comment:
Right now, a double update of the file object glitches the view. I'll check if it can be done in one API call instead.